### DISCUSSION TITLE
Android App

### DISCUSSION THREAD

**Discussion by loneranger1982** [blakeblackshear/frigate#2788] [Ask A Question]
HI everyone,

Got to say I absolutely love frigate and having been using it for several months now, 

I have created a Android app to work along side the web version, Not sure if this is something the community is interested in or have a use, i am looking for a few beta testers to test the app out.

The app is still in very early stages but happy to hear some feedback or criticism.
![frigate App](https://user-images.githubusercontent.com/18580245/153483106-6d76af34-a892-460a-9c04-87919262bd20.png)
![frigate App 2](https://user-images.githubusercontent.com/18580245/153483118-164c6d42-68a5-4b8a-b424-244b48097cf4.png)

**Comment by Minglarn** [comment]
Holy Bananas... This is what my wifey would love! 
Please add me to beta testing..

**Comment by NickM-27** [comment]
That's funny, I recently started working on something similar: https://github.com/NickM-27/Frigate-Android

Yours seems farther along so perhaps I could help contribute.

**Comment by pjfian** [comment]
Add me to the list too please.

**Comment by loneranger1982** [comment]
just setting up on google play store will invite all once setup

**Reply by loneranger1982** [reply]
need email address to add people to the closed testing group.

**Reply by Firerouge** [reply]
Why not opensource and release apk here on github?

**Reply by loneranger1982** [reply]
not ready to release yet still testing want to ensure it will work on different devices and different frigate setups. Planning to release by end of March if I can get it fully tested by then.

**Reply by pcb1962** [reply]
happy to help with testing - pcb1962@gmail.com

**Reply by scooper1** [reply]
happy to test   scooper1818@gmail.com

**Comment by gulo101** [comment]
Add me as tester please

**Comment by pcb1962** [comment]
Martin, shall we discuss feedback here or would you prefer email?
Very impressed so far 👍🏻

**Reply by loneranger1982** [reply]
Feature request ideas etc happy to discuss here bugs an errors if you can email as don't want to overload this discussion. Hope @blakeblackshear don't mind this discussion on here

**Reply by Minglarn** [reply]
Cool, how far have you come with the app, do you want my email so you can make some invites?

**Reply by loneranger1982** [reply]
Give me your email happy to give you an invite more testers the better,
App is almost at release state some tidying up and error checking before release, have some ideas for more function at future releases but for now can view camera feeds and see events and clips.

**Reply by Minglarn** [reply]
Great! 
e-mail: minglarn at gmail.com

**Comment by gulo101** [comment]
Hi
Can you add me to testing too? gulo101 gmail.com

Thanks

**Comment by Minglarn** [comment]
Okey, been testing briefly.. Been working like a horse :)
Anyway...
There are some "must-have"... 
1. Ability to use WAN instead of LAN address. I'm using my frigate instance with my own domain protected with ngnix password.
2. Settings to go directly to EVENTS for all cams whenever you open the app. 
3. Own GitHub, where ideas and bug reports are easy to follow and report. 

Ideas:
1. Allowing the grid size eg 3 cams in a row. 
2. Event list per object type, not only per event. Like: "animal", "person", "car" etc. 
3. Ability to open the app from HA companion app. Example: Whenever I get a notification I would like to open the recording in FrigateCctv app. 
4. Notifications directly sent to FrigateCCTV app? Dont know if this is possible but would be a GREAT thing to have!
5. Update snapshots every 30sec? Adjustable from settings menu would be a nice thing. 

More ideas are on my list. Great job btw.. 👍 
Looking forward to the next version.

**Comment by loneranger1982** [comment]
Email will be fine 
Get Outlook for Android

**Comment by scho0ck** [comment]
Would love to test the app as well

**Comment by loneranger1982** [comment]
Hi all, I have released this on google play store today, please check out ant enjoy. I plan to add some of the features requested by the end of march if there is anything else people want added please let me know link below for android, apple to follow shortly
https://play.google.com/store/apps/details?id=com.shenobis.frigatecctv

**Reply by Minglarn** [reply]
Nice work! Keep it coming...
Hope i'll contribute in some way. .. Just tell me how. :)

**Reply by PrplHaz4** [reply]
> Hi all, I have released this on google play store today, please check out ant enjoy. I plan to add some of the features requested by the end of march if there is anything else people want added please let me know link below for android, apple to follow shortly https://play.google.com/store/apps/details?id=com.shenobis.frigatecctv

It feels like this listing does not do enough to distance itself from the "official" Frigate and Frigate+ brand.  Being closed source it also does not really seem aligned closely enough to the Frigate open source project to be using the brand directly.

Other apps I've seen adopt completely different naming and/or state that it is not affiliated in the description.

`Frigate Viewer - unofficial app for viewing cameras on your Frigate NVR instance`

**Reply by loneranger1982** [reply]
> > Hi all, I have released this on google play store today, please check out ant enjoy. I plan to add some of the features requested by the end of march if there is anything else people want added please let me know link below for android, apple to follow shortly https://play.google.com/store/apps/details?id=com.shenobis.frigatecctv
> 
> It feels like this listing does not do enough to distance itself from the "official" Frigate and Frigate+ brand. Being closed source it also does not really seem aligned closely enough to the Frigate open source project to be using the brand directly.
> 
> Other apps I've seen adopt completely different naming and/or state that it is not affiliated in the description.
> 
> `Frigate Viewer - unofficial app for viewing cameras on your Frigate NVR instance`

Hi
You are correct I should clearly state this is an unofficial app and will do this on the app page. I am not trying to take away from the hard work the community has put into this project

**Comment by gulo101** [comment]
Hi

I just installed it (updated the testing version) and it keeps crashing, it
wont even open. I did try to reinstall from scratch and cleared data/cache.
Then the app opened, however after I entered my server IP it crashed
again and won't open again.

I sent crash report

thanks

**Reply by loneranger1982** [reply]
OK will look at this not sure why it has crashed as nothing has changed from testing. Will look at the crash report see what's going on thanks for the feedback

**Reply by loneranger1982** [reply]
@gulo101 update has been approved, this should fix your issue

**Comment by gulo101** [comment]
Ok it's opening now. However I can only see one camera now without paying?
I have no problem buying the app (thanks Google Opinion rewards!) , but I
feel like little more testing/ features would be nice before having to buy.

Thanks!

**Comment by pcb1962** [comment]
Same for me, not crashing after the update.
I'll be happy to buy it when I can give it the public URL (and user/pass)
of my frigate instance - ie https://frigate.mydomain.net (I reverse proxy
frigate via nginx) - is this in the plans? I don't need the app at home, I
have plenty of other ways to view the cameras at home already but this app
would be very useful away from home. Thanks, Peter

**Reply by loneranger1982** [reply]
Hi, 
Will make this a feature request which I should be able to get sorted quickly, will probably send you a update on the test stream later this week. I am planning loads of new features over the coming months. I am also planning to donate some of the revenue to this project to support the great work blakeblackshear is doing.

**Reply by loneranger1982** [reply]
i have pushed an update to the testing group you should see this now link below if needed

https://play.google.com/apps/internaltest/4698765315468366664

**Reply by pcb1962** [reply]
Hi, I've given that some testing.  
If I disable http Auth on the proxy it works, but only if I connect initially on port 80, then nginx sends you a 301 response with the https address and you reconnect on port 443. If I enter port 443 in the app config it doesn't connect, returning you HTTP/1.1 400 Bad Request (The plain HTTP request was sent to HTTPS port). It doesn't allow me to enter the frigate address as https://frigate.mydomain.net so that the initial connection can be made with encryption.
So that works for me (without Auth), because the app connects first on port 80 and allows nginx to force the redirection, but not everyone would have that redirection set up and may require to hit port 443 with an https connection from the outset.
However the Auth isn't working, I think because you're sending the auth data on the initial connection to port 80, where it's ignored, and not after the redirection to port 443. 
Hope that makes some sense, this isn't really a field in which I have much expertise!
In general I really like the app, it's very fast and responsive and I really like being able to scroll through a whole day of events, unlike the HA app where I can only see the past few hours of events without further narrowing things down. Will be happy to buy it, an absolute bargain at the price you have set, if you can sort out the https auth thing.

**Reply by loneranger1982** [reply]
another update pushed to testing couple more features add which I have been working on should be be fully working now, you will need to allow port 1935 in your firewall to see livestreams

**Comment by Minglarn** [comment]
Ehhh? Just updated the app, and now there is only ONE cam and option to upgrade to PRO for 3 bucks.. 
Did I miss something?

**Comment by gulo101** [comment]
Sadly again seeing crash upon opening with the latest version. Just sent
bug report.

**Reply by loneranger1982** [reply]
all sorted now on the testing stream if you would like to test, also made some modifications to the amount of columns on the event and camera view if you could test.

Thank You

**Comment by gulo101** [comment]
Thanks ! the app opens again now. I tested seting event columns to 3 and it
works. The camera list only has 1 camera now, not having pro version, so
little harder to test 😀

**Comment by gulo101** [comment]
Ok, so I just purchased the pro version and now the app won't open again.
Sent the crash log.

**Reply by loneranger1982** [reply]
Really sorry, you can test on a couple of devices but will never catch all errors, believe i have found the issue and now deploying a new update fingers crossed

**Reply by loneranger1982** [reply]
also have you had a chance to check the https option??

**Reply by pcb1962** [reply]
I've tested https with Auth, seems to be good now, but I can't play live or recordings, will test further tomorrow, thanks

**Comment by gulo101** [comment]
It did start working actually for some reason, so I'm good for now. Thanks!

**Comment by loneranger1982** [comment]
For those not on the testing stream i have now released a update to the Production version on play store with many more update to follow slowly work through the feature requests if you can think of any thing not mentioned already let me know i will try and complete all feature request.

**Comment by gulo101** [comment]
Thanks,

I would love to see a single event view from all cameras filtered to one
object. So for example view of all "person" events from all cameras. So if
I want to see all people related movements around my house I don't have to
go through each camera individually. Thanks.

**Comment by rhatguy** [comment]
I'm running frigate 0.10.0-bfecee9 and have a one plus 8 pro android phone running android 11.  When I view events they play fine, but when I try to pull up the live view of a camera I get a completely black screen even though I see the time incrementing on the bottom of the screen as if the video should be playing.  I assume there are some differences in how live video is played vs playing an event?

**Reply by loneranger1982** [reply]
This could be linked to the cameras having audio, you could remove audio inn the Frigate Config or try using the VLC player in the app settings,, This is something I am working hard on but haven't been able to get to figure out yet.

**Reply by rhatguy** [reply]
VLC player does work but takes a long time to come up on my phone.  I'll try removing audio from Frigate and see if the native player works fro the live streams without audio.  THX.

**Comment by loneranger1982** [comment]
I have just pushed a update to the testing group which will allow filtering of events not ready for production yet as want to do more work if possible to give some feedback.

Also if possible could some of you provide some feedback and ratings on the play store.

Thank You

**Comment by gulo101** [comment]
Hello

When I click on any camera I now see {} on top,  but no filter?

Also in the left main menu I see "Person" but it just brings up an empty
screen if I click it.

**Reply by loneranger1982** [reply]
i haven't sorted the filters while in the camera events view just from the menu option in home screen want to check concept before moving forward, checking the basics work. Will be working on add the filters while in camera view over the next week.
I have pushed a new version with some bug fixes and filter should work correctly

**Comment by gulo101** [comment]
ALL - {person} filter now working for me from the sidebar, cool!

**Comment by loneranger1982** [comment]
Update pushed to tester, Google login is to add purchases to the testing group and for future features, filters now possible on all cameras and individual cameras. If there are no issues should be published to production by end of the week.

**Reply by loneranger1982** [reply]
I have now updated the production version which has improved layout for tablets and larger screens, will continue to improve this with more updates expected y the end of April.

**Comment by loneranger1982** [comment]
More improvements made, fixed a few issues as well as adding new features, you can now drag down to refresh. There is also a page to show you the HDD space used. Is there any other must have features people would like to see in this app.

**Comment by gulo101** [comment]
Thanks, can you check formatting?

**Comment by loneranger1982** [comment]
Evening all, 
looking for some feedback I have currently been looking into notifications and currently the only was i can see this working is if i host the MQTT server or bridge the events to my hosted MQTT server. I have the testing app setup to send notifications with a bridge MQTT server this would require the end user to add a few lines of config to the Mosquitto MQTT broker to forward events to my server to then allow notifications.

Would people be happy to change the config for notifications?

Any better ideas?

iI should have the next release with notifications available by the end of may and for the PRO users at the beginning of May

Many Thanks

**Comment by dcseng** [comment]
Please, make this work on Android TV!! Would be fantastic!!

**Reply by loneranger1982** [reply]
Once i have figured out the notifications will definitely get this working on Android TV

**Reply by snickers2k** [reply]
+1 for androidTV
i dream for years of a picture-in-picture window that pops up as soon as someone shows up on the camera's

**Comment by Buster14** [comment]
Hi, any chance to implement timeline?

https://github.com/blakeblackshear/frigate/pull/2830

Big names like milestone xprotect have this feature, and I'm keeping a windows VM just for this feature.
Would be great if frigate can do this on android, most non tech savvy people (my wife, parents) relies on android for cctv footage. And they prefer timeline over event based GUI like blue iris.
Thanks

**Comment by mdezzi** [comment]
Hi, just installed this app and i am trying to get it connected to Frigate installation. I have Frigate reverse proxied and sitting behind an Authelia 2FA SSO login. After i entered my server address (https://frigate.domain.tld) the app force closes each time i update it. 

I assume the 2FA login is causing the app to crash. I disabled authelia and was able to log in, then re-enabled authelia. This should keep me active for 30 days until i have to do it again. It would be nice to catch that error and open the page in a web browser to allow to enter the 2FA. 

Edit: i take that back, once Authelia restarted and enabled the 2FA again the app crashes upon startup consistently.

**Comment by loneranger1982** [comment]
Thank you for highlighting the issue will investigate and try to catch the error and apply the 2fa.I don't know aithelia very well is there any options to add an exception to allow the app without?Get Outlook for Android

**Comment by mdezzi** [comment]
I'm not sure, but there appears to be a lot of discussion around the HA
companion app and trying to get through Authelia 2FA as well:
https://github.com/home-assistant/android/issues/1438

**Comment by guerrilladynamicsinc** [comment]
Please sign me up as a tester

**Comment by papka1974** [comment]
Installed app. Configured Nginx reverse proxy with https, added auth. Opened port on router. Checked that could connect and view interface from outside. Set all parameters within app and getting "Connection error: socket timeout.". Same moment everything working from web. What i'm doing wrong?

**Comment by n8091977** [comment]
I am struggling with one camera out of 5, I can not get live stream working via the app. All works ok via Home Assistant app. Is there a way to debug / look at logs for the app itself as the server side seems ok. I do get sound on live stream, just no video (its a doorbell camera). I can view events and recorded video for all cameras without issue.

**Comment by k-brok** [comment]
Hi, I like frigate a lot and wanted to use the app. Willing to pay for the pro version with my opinion reward. However apparently i'm not able to do so. You know why?

**Comment by ghost** [comment]
Installed app, connected to my frigate server via nginx proxy and ssl.  I can see the events for my camera, but no live view, and no video of the events (just the thumbnails).  Juts get a black screen, no error messages.  Cameras are all working fine on PC via web browser and nginx proxy.

Is there an error log somewhere?

Happy to pay for the app if it works.

Motorola MotoG 82 5G with all latest updates.

**Reply by pcb1962** [reply]
I found the same, couldn't get it working via my nginx https proxy, only when connected directly to the frigate host.

**Comment by MRobi1** [comment]
Would love to try this out, but play store won't let me install as it says it's for an older version of Android.

**Comment by Systemedic** [comment]
Is this still active? If yes can I be included in the test?

**Reply by dnestico** [reply]
Would love to be included too! Thanks

**Comment by KickingAss2024** [comment]
Well is this becoming Open source?
